chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1290 @@
+pub enum ChannelPin {
+    A,
+    B,
+}
+
+/// Check ChannelNumber implementation for more details
+impl From<RPGpio> for ChannelPin {
+    fn from(gpio: RPGpio) -> Self {
+        match gpio as u8 & 0b0000_0001 {
+            // Because of the bitwise AND, there are only two possible values
+            0 => ChannelPin::A,
+            _ => ChannelPin::B,
+        }
+    }
+}
+
+/// PWM channel configuration structure
+///
+/// This helper struct allows multiple channels to share the same configuration.
+struct PwmChannelConfiguration {
+    en: bool,
+    ph_correct: bool,
+    a_inv: bool,
+    b_inv: bool,
+    divmode: DivMode,
+    int: u8,
+    frac: u8,
+    cc_a: u16,
+    cc_b: u16,
+    top: u16,
+}
+
+impl Default for PwmChannelConfiguration {
+    /// Create a set of default values to use for configuring a PWM channel:
+    /// + the channel is disabled
+    /// + trailing-edge modulation configured
+    /// + no pin A and B polarity inversion
+    /// + free running mode for the fractional clock divider
+    /// + integral part of the divider is 1 and the fractional part is 0
+    /// + compare values for both pins are set 0 (0% duty cycle)
+    /// + top value is set to its maximum value
+    fn default() -> Self {
+        PwmChannelConfiguration {
+            en: false,
+            ph_correct: false,
+            a_inv: false,
+            b_inv: false,
+            divmode: DivMode::FreeRunning,
+            int: 1,
+            frac: 0,
+            cc_a: 0,
+            cc_b: 0,
+            top: u16::MAX,
+        }
+    }
+}
+
+const PWM_BASE: StaticRef<PwmRegisters> =
+    unsafe { StaticRef::new(0x40050000 as *const PwmRegisters) };
+
+/// Main struct for controlling PWM peripheral
+pub struct Pwm<'a> {
+    registers: StaticRef<PwmRegisters>,
+    clocks: OptionalCell<&'a clocks::Clocks>,
+    interrupt_handler: OptionalCell<&'a dyn Interrupt>,
+}
+
+impl<'a> Pwm<'a> {
+    /// Create a new Pwm struct
+    ///
+    /// **Note**:
+    /// + This method must be called only once when setting up the kernel peripherals.
+    /// + This peripheral depends on the chip's clocks.
+    /// + Also, if interrupts are required, then an interrupt handler must be set. Otherwise, all
+    /// the interrupts will be ignored.
+    ///
+    /// See [Pwm::set_clocks]
+    pub fn new() -> Self {
+        let pwm = Self {
+            registers: PWM_BASE,
+            clocks: OptionalCell::empty(),
+            // There is no option to create an array of OptionalCell, so
+            // only one global interrupt_handler is supported.
+            // If arrays of OptionalCell are going to be added,
+            // then it will be possible to configure an interrupt handler
+            // per PWM channel and provide a more user-friendly API.
+            interrupt_handler: OptionalCell::empty(),
+        };
+        pwm.init();
+        pwm
+    }
+
+    // enable == false ==> disable channel
+    // enable == true ==> enable channel
```

Comment:
```suggestion
    /// Describe the function
    ///
    /// enable == false ==> disable channel
    /// enable == true ==> enable channel
```
Please write the documentation in rustdoc format

Reply:
Changed comments, although I thought only public functions should be included in the documentation.

Reply:
Yes, you are right, please use `//`. Sorry for this.